### ISSUE TITLE
`any` with `sleep`

### ISSUE THREAD

**Issue by lucasmreis** [bbarr/medium#3]
First of all, nice work with the library! I really like the API, seems the best from all javascript CSP implementations I've come across. I really like `repeat` :)

I'm having an issue with the `any` function: whenever I use it with a promise as a parameter (say `sleep(100)`), the function simply does not work. I'm running this code:

``` js
import 'babel-polyfill'
import { chan, put, sleep, go, any } from 'medium'

const channel = chan()

go(async () => {
  const ch1 = chan()
  const p = sleep(100)

  put(ch1, 123)

  const [ v, ch ] = await any(ch1, p)

  console.log('Result:', v)
})
```

And nothing is printed. If I remove the promise `p` from the call, I get the result printed:

``` js
import 'babel-polyfill'
import { chan, put, sleep, go, any } from 'medium'

const channel = chan()

go(async () => {
  const ch1 = chan()
  const p = sleep(100)

  put(ch1, 123)

  const [ v, ch ] = await any(ch1) // p removed

  console.log('Result:', v)
})
```

```
> Result: 123
```

I've run the tests locally, and they all pass, which is strange. 

I'm using babel 6.5, both compiling the code and running `node`, and using `babel-node`, both on Node 4.2.4 and 5.6.0.

Can anyone reproduce this?

**Comment by bbarr** [comment]
Hmm, unfortunately, I can't reproduce this. I just bumped babel to 6.5 (node -v 5.6.0) and added a test with the exact code that doesn't work for you and it all passes.

For the next step, could you try pulling down these updates, run the tests and see if things are still green for you? If they are, problem should be solved, if not, post any failures.

And thanks for the kind words, @lucasmreis :)

**Comment by lucasmreis** [comment]
Thanks for the attention!

The tests run ok. I'm running this even simpler code now:

``` js
import 'babel-polyfill'

import { chan, put, sleep, go, any } from 'medium'

const ch1 = chan()
const p = sleep(100)

put(ch1, 123)

any(ch1, p).then(x => console.log(x))
```

And I get this more informative error output:

```
/Users/lucasmreis/Projects/async-csp-exp/node_modules/medium/build/index.js:242
    return !ch.buffer.isEmpty();
                     ^

TypeError: Cannot read property 'isEmpty' of undefined
    at /Users/lucasmreis/Projects/async-csp-exp/node_modules/medium/build/index.js:242:22
    at Array.filter (native)
    at any (/Users/lucasmreis/Projects/async-csp-exp/node_modules/medium/build/index.js:241:19)
    at Object.<anonymous> (/Users/lucasmreis/Projects/async-csp-exp/out.js:28:17)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:140:18)
```

It points to the babel output, that transpiled the `let alreadyReady = ports.filter(isResolvable)` expression.

It compiles as:

``` js
var ready = chs.filter(function (ch) {
  return !ch.buffer.isEmpty();
});
```

And, because `sleep` is a promise and does not have a property `buffer`, it breaks.

Changing the compiled code to:

``` js
var ready = chs.filter(function (ch) {
  if (typeof ch.then === 'function') {
    return false;
  }
  return !ch.buffer.isEmpty();
});
```

everything works fine!

So, it seems that something strange is happening in the transpilation phase, and your test for promise in `isResolvable` is getting thrown away.

**Comment by bbarr** [comment]
Ahem. So this is where I hide my head in shame.. I noticed that your compiled code didn't look right, and it turns out, because I had my tests pointing at src/ instead of build/, I was able to push / npm publish a version that DIDN'T include updated build files. 

I have fixed this, now running tests against only what is built, so this shouldn't happen again. Also, bumped version to 0.5.0 and republished module, as there were some breaking changes. Go ahead and update to this version, and things should "just work". Sorry about that!, and let me know if this does indeed resolve the issue.
